### PR TITLE
Update to design document overview and trim of appendix

### PR DESCRIPTION
**Pull request by spencerkimball**
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/9648)

### CODE REVIEW

File: docs/design.md [hunk 1]
```diff
@@ -4,42 +4,65 @@ by Spencer Kimball from early 2014.
...
-
-A single logical mutation may affect multiple key/value pairs. Logical
-mutations have ACID transactional semantics. If all keys affected by a
-logical mutation fall within the same range, atomicity and consistency
-are guaranteed by Raft; this is the **fast commit path**. Otherwise, a
-**non-locking distributed commit** protocol is employed between affected
-ranges.
+CockroachDB is a distributed SQL database. The primary design goals
+are **scalability**, **strong consistency** and **survivability**
+(hence the name). Cockroach aims to tolerate disk, machine, rack, and
```
Comment (petermattis) [inline docs/design.md]: Should we consistently use `CockroachDB` in this document?

File: docs/design.md [hunk 2]
```diff
@@ -4,42 +4,65 @@ by Spencer Kimball from early 2014.
...
+and values are byte strings.
+
+The KV map is logically composed of smaller segments of the keyspace
+called ranges. Each range is backed by data stored in a local KV
+storage engine (we use [RocksDB](http://rocksdb.org/), a variant of
+LevelDB). Range data is replicated to a configurable number of
+additional CockroachDB nodes. Ranges are merged and split to maintain
+a target size, by default `64M`. The relatively small size facilitates
+quick repair and rebalancing to address node failures, new capacity
+and even read/write load.
```
Comment (petermattis) [inline docs/design.md]: There is also pressure for a larger size to minimize the `meta2` index size.

File: docs/design.md [hunk 3]
```diff
@@ -4,42 +4,65 @@ by Spencer Kimball from early 2014.
...
+
+CockroachDB achieves horizontally scalability:
+- adding more nodes increases the capacity of the cluster by the
+  amount of storage on each node (divided by a configurable
+  replication factor), theoretically up to 4 exabytes (4E) of logical
+  data;
+- client queries can be sent to any node in the cluster, and queries
+  can operate fully independently from each other, meaning that
```
Comment (petermattis) [inline docs/design.md]: The queries can operate independently when there aren't conflicts. Not sure if that detail is worth calling out here.

File: docs/design.md [hunk 4]
```diff
@@ -4,42 +4,65 @@ by Spencer Kimball from early 2014.
...
+  overall throughput is a linear factor of the number of nodes in the
+  cluster.
+- queries are distributed (ref: distributed SQL) so that the overall
+  throughput of single queries can also be increased by adding more
```
Comment (petermattis) [inline docs/design.md]: `s/can also be/can be/g`

File: docs/design.md [hunk 5]
```diff
@@ -4,42 +4,65 @@ by Spencer Kimball from early 2014.
...
+  nodes.
+
+CocroachDB achieves strong consistency:
```
Comment (petermattis) [inline docs/design.md]: `s/CocroachDB/CockroachDB/g`

File: docs/design.md [hunk 6]
```diff
@@ -4,42 +4,65 @@ by Spencer Kimball from early 2014.
...
+- single or batched mutations to a single range are mediated via the
+  range's Raft instance. Raft guarantees ACID semantics.
+- logical mutations which affect multiple ranges employ distributed
+  transactions for ACID semantics. CockroachDB uses an efficient
+  **non-locking distributed commit** protocol.
+
+CockroachDB achieves survivability:
+- range replicas can be co-located within a single datacenter for low
+  latency replication and survive disk or machine failures. They can
+  be located across racks to survive some network switch failures.
```
Comment (petermattis) [inline docs/design.md]: `s/located/distributed/g`

File: docs/design.md (hunk 1 above)
Comment (spencerkimball) [inline docs/design.md]: Done

File: docs/design.md (hunk 2 above)
Comment (spencerkimball) [inline docs/design.md]: Mentioned.

File: docs/design.md (hunk 3 above)
Comment (spencerkimball) [inline docs/design.md]: Done

File: docs/design.md (hunk 4 above)
Comment (spencerkimball) [inline docs/design.md]: Done

File: docs/design.md (hunk 5 above)
Comment (spencerkimball) [inline docs/design.md]: Done

File: docs/design.md (hunk 6 above)
Comment (spencerkimball) [inline docs/design.md]: Done

**Review by spencerkimball — COMMENTED** [review]
Submitting as a separate PR

**Pull request merged**